FreeRTOS/Test/CMock/makefile.in

```diff
@@ -17,26 +17,34 @@ FREERTOS_DIR     := $(abspath $(FREERTOS_DIR_REL))
 KERNEL_DIR_REL   := ../../../FreeRTOS/Source
 KERNEL_DIR       := $(abspath $(KERNEL_DIR_REL))
 
-CMOCK_DIR        := $(UT_ROOT_DIR)/CMock
-CMOCK_SRC_DIR    := $(CMOCK_DIR)/src
-UNITY_DIR        := $(CMOCK_DIR)/vendor/unity
-UNITY_SRC_DIR    := $(UNITY_DIR)/src
-UNITY_INC_DIR    := $(UNITY_DIR)/src
-UNITY_BIN_DIR    := $(UNITY_DIR)/auto
+CMOCK_DIR           := $(UT_ROOT_DIR)/CMock
+CMOCK_SRC_DIR       := $(CMOCK_DIR)/src
+UNITY_DIR           := $(CMOCK_DIR)/vendor/unity
+UNITY_SRC_DIR       := $(UNITY_DIR)/src
+UNITY_INC_DIR       := $(UNITY_DIR)/src
+UNITY_BIN_DIR       := $(UNITY_DIR)/auto
+UNITY_EXTRAS_DIR    := $(UNITY_DIR)/extras
+UNITY_MEM_SRC_DIR   := $(UNITY_EXTRAS_DIR)/memory/src
 
 CMOCK_EXEC_DIR   := $(CMOCK_DIR)/lib
 
 # Include directory location
 INCLUDE_DIR      := -I$(KERNEL_DIR)/include -I. -I$(UT_ROOT_DIR)/config
 INCLUDE_DIR      += -I$(UNITY_INC_DIR)
 INCLUDE_DIR      += -I$(CMOCK_SRC_DIR)
+INCLUDE_DIR      += -I$(UNITY_MEM_SRC_DIR)
 
 CPPFLAGS :=
 CFLAGS   := $(INCLUDE_DIR)  -O0 -ggdb -pthread  --std=c99 -Werror -Wall
-LDFLAGS  := -L$(LIB_DIR) -lunity -lcmock -Wl,-rpath,$(LIB_DIR) -pthread -lgcov
+LDFLAGS  := -L$(LIB_DIR) -Wl,-rpath,$(LIB_DIR) -fsanitize=address
```

Comment:
I am thinking if it would be nice to add different sanitizers with different targets for example:
make run_sanitize_address
make run_sanitize_memory
ect
what do you think ?

Reply:
Why not run all the sanitizers each time?

Reply:
since some sanitizers cannot run together if i remember correctly

Reply:
-fstack-check=' and '-fstack-clash_protection' are mutually exclusive, but accomplish the same goal.